source/material_model/rheology/fk_viscosity.cc

```diff
@@ -0,0 +1,125 @@
+/*
+  Copyright (C) 2019 - 2020 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/material_model/rheology/fk_viscosity.h>
+#include <aspect/utilities.h>
+
+#include <deal.II/base/signaling_nan.h>
+#include <deal.II/base/parameter_handler.h>
+
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    namespace Rheology
+    {
+      template <int dim>
+      FkViscosity<dim>::FkViscosity ()
+      {}
+
+
+
+      template <int dim>
+      double
+      FkViscosity<dim>::compute_viscosity (const double ref_viscosity,
+                                           const double temperature,
+                                           const unsigned int composition) const
+      {
+        // Frank-Kamenetskii viscosity approximation
+        //    viscosity = ref_v * A * B * exp(E * (0.5-(T/ref_T)/2)))
+        // ref_v: reference viscosity,
+        // A: prefactor, B: pre-limiter, E: adjusted viscosity ratio,
+        // ref_T: reference temperature, T: temperature.
+        double viscosity_fk = 0.0;
+	      if (depth > 660e3) // I'm not sure of the best way to access depth here
+	        viscosity_fk = ref_viscosity * fk_visc_pre_lm[composition] * fk_visc_pre[composition] * std::exp(fk_visc_E[composition]*(0.5-temperature/reference_T/2));
+        else
+          viscosity_fk = ref_viscosity * fk_visc_pre[composition] * std::exp(fk_visc_E[composition]*(0.5-temperature/reference_T/2));
+
+        return viscosity_fk;
```

Comment:
Lets talk about how to formulate this exactly. We have an alternative version in the `LatentHeat` material model (see for example #3693), and the question of phase transitions (e.g. upper mantle / lower mantle) is handled in #3598. We should sit together and talk about how to combine this for this module.